hist/histv7/doc/Terminology.md

```diff
@@ -0,0 +1,53 @@
+# Histogram Terminology
+
+This document collects, defines, and explains terms that are used in ROOT's histogram package.
+The goal is to start from a common understanding, which should avoid ambiguities and ease discussions.
+It also helps (future) developers to navigate the code because classes and methods are named accordingly.
+The list is ordered alphabetically, though dependent terms are kept together with their parent.
+It is supposed to be exhaustive; any missing term should be added when needed.
+
+An *axis* is a bin configuration in one dimension.
+A *regular axis* has equidistant bins in the interval $[a, b)$.
+A *variable bin axis* is configured with explicit bin edges $[e_{n}, e_{n+1})$.
+A *categorical axis* has a unique label per bin.
+*Axes* is the plural of axis and usually means the bin configurations for all dimensions of a histogram.
+
+A *bin content* is the value of a single bin.
+The *bin content type* can be an integer type, a floating-point type, the special `DoubleBinWithError`, or a user-defined type.
+
+A *bin error* is the Poisson error of a bin content.
+With the special `DoubleBinWithError`, it is the square root of the sum of weights squared: $\sqrt{\sum w_i^2}$
+Otherwise it is the square root of the bin content, which is only correct with unweighted filling.
+
+A *bin index* (plural *indices*) refers to a single bin of a dimension.
```

Comment:
What does this mean for multi-dimensional histograms?

Reply:
For a multi-dimensional histogram, you'll need an array of bin indices to refer to a single bin content. It's indeed ambiguous how we define a "bin": Is it per dimension or is it only once you give coordinates for all dimensions, ie what really has the bin content in the end?

Reply:
The ambiguity theoretically exists also for the TH1-style histograms, but in practice, it is always clear from the context if you are talking about a bin on an axis (1-dim), or a bin in the histogram (possibly an index tuple). I guess one could add a sentence about bins in histogram and cover all cases.

Reply:
Yes, that's why there is no definition of a "bin" itself :innocent: for "bin content" it is clear that it's the bin of a histogram, and for a single "bin index" it makes life easier if that's on a single axis.

Could you concretely propose where you would like to see added what?